.circleci/config.yml

```diff
@@ -1,37 +1,320 @@
+
+# Attaches the release output which has been stored in the workspace to the current job.
+# https://circleci.com/docs/2.0/workflows/#using-workspaces-to-share-data-among-jobs
+var_9: &attach_release_output
+  attach_workspace:
+    at: dist/releases
+
+# -----------------------------
+# Container version of CircleCI
+# -----------------------------
 version: 2
+
+# -----------------------------------------------------------------------------------------
+# Job definitions. Jobs which are defined just here, will not run automatically. Each job
+# must be part of a workflow definition in order to run for PRs and push builds.
+# -----------------------------------------------------------------------------------------
 jobs:
+
+  # -----------------------------------
+  # Build job
+  # -----------------------------------
   build:
-    docker:
-      # specify the version you desire here
-      - image: circleci/node:8.9
+    <<: *job_defaults
+    resource_class: xlarge
+    steps:
+      - *checkout_code
+      - *restore_cache
+      - *yarn_install
+
+      - run: yarn gulp :publish:build-releases
+
+      # Note: We want to save the cache in this job because the workspace cache also
+      # includes the build cache that will be updated in this job.
+      - *save_cache
+
+  # ------------------------------------------------------------------------------------------
+  # Job that runs the unit tests on locally installed browsers (Chrome and Firefox headless).
+  # The available browsers are installed through the angular/ngcontainer Docker image.
+  # ------------------------------------------------------------------------------------------
+  tests_local_browsers:
+    docker: *docker-firefox-image
+    resource_class: xlarge
+    steps:
+      - *checkout_code
+      - *restore_cache
+      - *yarn_install
+
+      - run: ./scripts/circleci/run-local-browser-tests.sh
+
+  # ----------------------------------------------------------------------------
+  # Job that runs the unit tests on Browserstack. The browsers that will be used
+  # to run the unit tests on Browserstack are set in: test/browser-providers.js
+  # ----------------------------------------------------------------------------
+  tests_browserstack:
+    <<: *job_defaults
+    resource_class: xlarge
+    environment:
+      BROWSER_STACK_USERNAME: "adamplumer1"
+      BROWSER_STACK_ACCESS_KEY: "WgLjxoB2zQ3tqmsznKnz"
+    steps:
+      - *checkout_code
+      - *restore_cache
+      - *yarn_install
+
+      - run: ./scripts/circleci/run-browserstack-tests.sh
+
+  # ----------------------------------------------------------------------------
+  # Job that runs the unit tests on Saucelabs. The browsers that will be used
+  # to run the unit tests on Saucelabs are set in: test/browser-providers.js
+  # ----------------------------------------------------------------------------
+  tests_saucelabs:
+    <<: *job_defaults
+    resource_class: xlarge
```

Comment:
Flex-layout should probably go down with the resource class here. On Material we have it enabled for tests because we have around 3k. `Medium` is the default and should be good enough, thoughts?

(same applies for browserstack, local browsers and saucelabs)

Reply:
I kind of like that it runs super-quick though 😉 . But you're probably right.  The default should probably be plenty. I'll try it out in a fixup commit, and if it's not good enough I'll revert back